src/api/offers.js

```diff
@@ -47,15 +56,16 @@ export default ({offerService}) => {
     param('token').isHexadecimal().isLength(TOKEN_LENGTH),
     validateMiddleware,
     async (req, res) => {
-      const {token} = matchedData(req)
-      const offer = offerService.findOfferByToken(token)
-      if (!offer) {
-        return res.status(HttpStatus.NOT_FOUND).json('No such offer')
-      } else if (offer.isExpired() || offer.deleted) {
-        return res.status(HttpStatus.GONE).json('Offer not available')
-      }
-
       try {
+        const {token} = matchedData(req)
+        const offer = new Offer(await offerService.getOfferByToken(token))
+
+        if (!offer) {
+          return res.status(HttpStatus.NOT_FOUND).json('No such offer')
```

Comment:
Does return a JSON object? If yes do we have a defined format for error codes? Like:
```
{
    "message": "Error message",
    "code": 55
    "additional_data": {}
}
```

Reply:
Not yet, should we establish one?